cwe_checker_rs/src/analysis/backward_interprocedural_fixpoint/mock_context.rs

```diff
@@ -0,0 +1,126 @@
+use super::*;
+use crate::analysis::graph::Graph;
+use petgraph::graph::NodeIndex;
+use std::collections::{HashMap, HashSet};
+
+/// Identifier for BlkStart and BlkEnd nodes
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum StartEnd {
+    Start,
+    End,
+}
+
+/// A simple mock context, only containing the program cfg
+#[derive(Clone)]
+pub struct Context<'a> {
+    pub graph: Graph<'a>,
+    pub tid_to_node_index: HashMap<(Tid, Tid, StartEnd), NodeIndex>,
+}
+
+impl<'a> Context<'a> {
+    pub fn new(project: &'a Project) -> Self {
+        let mut graph = crate::analysis::graph::get_program_cfg(&project.program, HashSet::new());
+        let mut tid_to_node_index: HashMap<(Tid, Tid, StartEnd), NodeIndex> = HashMap::new();
+        graph.reverse();
```

Comment:
I almost missed this one. My hope was that the user does not have to explicitly reverse the control flow graph when defining a context for a backwards analysis, so that forward and backward analyses could be performed on the same control flow graph. We should check together whether we can achieve this somehow.

Reply:
We could simply add a parameter 'backward: bool' to the get_program_cfg() and reverse the graph in that function. However, the user would have to add that parameter as well.